front/lib/plans/subscription.ts

```diff
@@ -0,0 +1,247 @@
+import { Op } from "sequelize";
+
+import { front_sequelize, ModelId } from "@app/lib/databases";
+import { Plan, Subscription } from "@app/lib/models";
+import {
+  FREE_TEST_PLAN_DATA,
+  FREE_UPGRADED_PLAN_CODE,
+  PlanAttributes,
+} from "@app/lib/plans/free_plans";
+import { generateModelSId } from "@app/lib/utils";
+import logger from "@app/logger/logger";
+import { SubscribedPlanType } from "@app/types/user";
+
+export const getActiveWorkspacePlan = async ({
+  workspaceModelId,
+}: {
+  workspaceModelId: ModelId | null;
```

Comment:
Any chance we could not rely on ModelId here?

I really hate that we use ModelId in the signature of that function. Maybe it should be internalized in the workspace loading code of auth?

Reply:
Problem is in `Authenticator` takes a workspace that comes from a db query findOne, so it's a `Workspace` not a `WorkspaceType`. Do you want me to refactor that?

Reply:
This PR is the epitome of inverted 80/20 and I'm not convinced we reached the best design yet (and might be wrong, but would love to thoroughly review the different possibilities with you IRL)

Reply:
(the fact that we have to load all Subscriptions for all workspace of a user when we load the `User` tells me that the design is not final)

Reply:
(sequentially or not, even if sequentially is terribly inefficient for something we call everywhere always)

Reply:
You are right @spolu . I've taken out plan out of workspace and it looks better. 
But this PR is a mess. I am closing it and will open multiple ones tonight with hopefully better value. 🙏🏻

Reply:
ack :+1: